packages/core/src/helpers/utilities.ts

```diff
@@ -0,0 +1,32 @@
+type CacheDecorator = {
+    (): ((target: unknown, propertyKey: string, descriptor: PropertyDescriptor) => void);
+    cache: Map<string, unknown>;
+};
+
+export function CacheDecoratorFactory(): CacheDecorator {
+    const cache = new Map<string, unknown>();
+    const decorator = function CacheDecorator() {
+        return function (_: unknown, propertyKey: string, descriptor: PropertyDescriptor): void {
```

Comment:
What's the use of this first parameter?

Reply:
Nevermind. It is using the decorator parameters. Therefore, it's the target, and you've named it "_" probably not to give it too much importance.

Reply:
I've simplified it after our conversation: 0982740f1a685bc760810bd238bdf4dd69800490